apps/store/src/components/Perils/Perils.tsx

```diff
@@ -0,0 +1,112 @@
+import styled from '@emotion/styled'
+import React, { useState } from 'react'
+import { CrossIcon } from 'ui'
+import * as Accordion from '@/components/Accordion/Accordion'
+import { SpaceFlex } from '@/components/SpaceFlex/SpaceFlex'
+import { CheckIcon } from './CheckIcon'
+import { MinusIcon } from './MinusIcon'
+import { PlusIcon } from './PlusIcon'
+
+type Peril = {
+  icon: React.ReactNode
+  name: string
+  description: string
+  covered: Array<string>
+  notCovered: Array<string>
+}
+
+type PerilsProps = {
+  perils: Array<Peril>
+}
+
+export const Perils = ({ perils }: PerilsProps) => {
+  const [openedItems, setOpenedItems] = useState<Array<string>>()
```

Comment:
I know we're talking about low number of items in most cases, but using Set would be more efficient.  ```openedItems.includes``` on every item means quadratic complexity

Reply:
🤔 I see. I understand that [`Set.has` is faster than `Array.prototype.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#performance) but I guess on this specific case adding it might just complicate things since we don't have a performance issue yet. But maybe I've missing something. The following is how I understood how a `Set` would help:

```suggestion
  const [openedItems, setOpenedItems] = useState<Set<string>>(new Set())

  const handleValueChange = useCallback((value: Array<string>) => {
    setOpenedItems(new Set(value))
  }, [])

  return (
    <Accordion.Root
      type="multiple"
      value={Array.from(openedItems)}
      onValueChange={handleValueChange}
    >
        {perils.map(({ icon, name, description, covered, notCovered }) => {
            const isOpened = openedItems.has(name)
```

As one can see, I'd need to create a new `Set` so React re-renders the component and then convert that `Set` into a regular array in order to be compliance with `Accordion.Root` API. If I got correctly we'd be moving from O(n^2) --> O(n), which is fair. In any case, I don't mind adding it :)

Reply:
Forgot we need to pass down value as well.  to/from conversion on every change does not seem ergonomic.   

It's easier to keep state an array and use set only for `isOpen`. Something like
```
const openedSet = new Set(openedItems)

perils.map((...) => {
  const isOpened = openedSet.has(name)
})
```